helpers/metadata/backends/parquet.py

```diff
@@ -136,6 +136,12 @@ def _extract_captions_to_fast_list(self):
                 filename = str(row[filename_column])
             else:
                 filename = str(index)
+            if type(caption_column) == list:
+                caption = None
+                if len(caption_column) > 0:
+                    caption = [row[c] for c in caption_column]
+            else:
+                caption = row[caption_column]
```

Comment:
Do we wanna randomize the caption sampling part similar to [this](https://github.com/huggingface/diffusers/blob/4ad7a1f5fd48ce3aabc46f090f28b56a02e733aa/examples/text_to_image/train_text_to_image.py#L780)?

Reply:
this job is to retrieve all captions for caching